### DISCUSSION TITLE
Regarding the problem of contact angle being reset by regenerate_models

### DISCUSSION THREAD

**Discussion by livashu** [PMEAL/OpenPNM#2782] [Q&A] [answered]
After manually adding the contact angle, regenerate_models for recalculation will force the contact angle to 110 degrees. How can I recalculate the contact angle with the manually changed value? 

Here is a simplified text.

Input:
import openpnm as op
import numpy as np
op.visualization.set_mpl_style()

pn = op.network.Cubic([20, 20, 20], spacing=5e-5)

water = op.phase.Water(network=pn)
water['pore.contact_angle']=150
water['pore.temperature']=298
water.add_model_collection(op.models.collections.phase.water
print(water['pore.contact_angle'])

water.regenerate_models()
print(water['pore.contact_angle'])

Output:
[150 150 150 ... 150 150 150]
[110 110 110 ... 110 110 110]

**Comment by jgostick** [comment, marked as answer]
Just delete the contact angle model I think, with `del water.models['throat.contact_angle']`.

**Reply by livashu** [reply]
Resolved. Thank you very much!
